server/src/main/java/io/crate/planner/optimizer/rule/RewriteFilterOnOuterJoinToInnerJoin.java

```diff
@@ -294,4 +295,15 @@ public Object get(int index) {
             return null;
         }
     };
+
+    private static class RuleSymbolEvaluator extends SymbolEvaluator {
```

Comment:
Could extend this to handle references and fields as well and remove:

https://github.com/crate/crate/blob/a3eb7f078f1e7981b7489e03ecf93d027bb575b0/server/src/main/java/io/crate/planner/optimizer/rule/RewriteFilterOnOuterJoinToInnerJoin.java#L276-L279

That way, all of this would be in a single place. (We'll need OuterColumn handling for correlated joins as well)

Reply:
Thx for the suggestion!